admin_guide/scheduling/descheduler.adoc

```diff
@@ -0,0 +1,302 @@
+====
+
+The descheduler job and descheduler pod are created in `kube-system` project.
+
+[IMPORTANT]
+====
+The descheduler is a Technology Preview feature only.
+ifdef::openshift-enterprise[]
+Technology Preview features are not supported with Red Hat production service
+level agreements (SLAs), might not be functionally complete, and Red Hat does
+not recommend to use them for production. These features provide early access to
+upcoming product features, enabling customers to test functionality and provide
+feedback during the development process.
+
+For more information on Red Hat Technology Preview features support scope, see
+https://access.redhat.com/support/offerings/techpreview/.
+endif::[]
+====
+
+The descheduler does not evict the following types of pods:
+
+* Critical pods (with the `scheduler.alpha.kubernetes.io/critical-pod` annotation).
+* Pods (link:https://kubernetes.io/docs/tasks/administer-cluster/static-pod/[static and mirror pods] or pods in standalone mode) not associated with a Replica Set (RS), Replication Controller (RC), Deployment, or Job (because these pods are not recreated).
+* Pods associated with xref:../../dev_guide/daemonsets.adoc#dev-guide-daemonsets[DaemonSets].
+* Pods with local storage.
+* Pods subject to xref:../../admin_guide/managing_pods.html#managing-pods-poddisruptionbudget[Pod Disruption Budget (PDB)] 
+are not evicted if descheduling violates the PDB. The pods can be evicted using 
+an [xref:../../admin_guide/out_of_resource_handling.html#out-of-resource-eviction-policyeviction policy].
+
+[NOTE]
+====
+Best efforts pods are evicted before Burstable and Guaranteed pods.
+====
+
+[[admin-guide-descheduler-role]]
+== Creating a Cluster Role
+
+To configure the necessary permissions for the descheduler to work in a pod: 
+
+. Create a ../../architecture/additional_concepts/authorization.adoc#roles[cluster role] with the following rules:
++
+[source,yaml]
+----
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: descheduler-cluster-role
+rules:
+- apiGroups: [""]
+  resources: ["nodes"]
+  verbs: ["get", "watch", "list"] <1>
+- apiGroups: [""]
+  resources: ["pods"]
+  verbs: ["get", "watch", "list", "delete"] <2>
+- apiGroups: [""]
+  resources: ["pods/eviction"] <3>
+  verbs: ["create"]
+----
+
+<1> Configures the role to allow viewing nodes.
+<2> Configures the role to allow viewing and deleting pods.
+<3> Allows a node to evict pods bound to itself
+
+. Create the xref:../../admin_guide/service_accounts.adoc#admin-guide-service-accounts[service account] which will be used to run the job:
++
+----
+# oc create sa <file-name>.yaml -n kube-system
+----
+
+. Bind the cluster role to the service account:
++
+----
+# oc create clusterrolebinding descheduler-cluster-role-binding \
+    --clusterrole=descheduler-cluster-role \
+    --serviceaccount=kube-system:descheduler-sa
```

Comment:
Since you hardcord the serviceaccount name but create sa prompt user to provide a name.

So please make it consistent.

Reply:
ignore this